synapse/federation/transport/client.py

```diff
@@ -388,6 +404,7 @@ async def send_leave_v1(
             # server was just having a momentary blip, the room will be out of
             # sync.
             ignore_backoff=True,
+            parser=LegacyJsonDictParser(),
```

Comment:
The v1 versions of `send_{leave,join}` have a weird return type of `[status code, actual response]`.